### PR TITLE
ECF Registration Marts

### PR DESCRIPTION
**Pull request by consult1815**
I've created four marts that underpin the ECF Registration Dashboard. The first covers the school related records and the other three are concerned with participant induction records (one for ECTs, one for Mentors, and one for both types of participants that can be used for graphs), these three use nearly identical logic.

### CODE REVIEW

File: definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_ect_inductions.sqlx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  -- #This table generates a list of the dates from May - October in each approximate registration window (May - October) #
+  registration_dates AS (
+SELECT
+  *
+FROM (
+  SELECT
+    DISTINCT(start_year) AS reg_year,
+    CAST(CONCAT(start_year,'-05-01') AS date) AS reg_start_date,
+    CAST(CONCAT(start_year,'-10-31') AS date) AS reg_end_date
+  FROM
```
Comment (AntPage1) [inline definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_ect_inductions.sqlx:53]: The FROM clause is not using the Javascript `${ref()}`

**Review by AntPage1 — CHANGES_REQUESTED** [review]
**ls_ecf_reg_ect_inductions:** Descriptions and comments are clear and make the code understandable. No errors identified or issues. Only area that needs attention are the FROM clauses for dataform tables/marts where the Javascript `${ref()}` isn't being used.

**ls_ecf_reg_mentor_inductions:** Table description doesn't quite match ls_ecf_reg_ect_inductions (see comment below) and the FROM clauses need to be changed to Javascript `${ref()}`

File: definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_mentor_inductions.sqlx [hunk 2]
```diff
@@ -0,0 +1,126 @@
+config {
+    type: "table",
+    assertions: {
+        uniqueKey: ["induction_record_id"]
+    },
+    bigquery: {
+        partitionBy: "DATE(created_at)",
+        clusterBy: ["cohort"]
+    },
+    description: "This mart pulls together all Mentor induction records generated within the registration periods from 2022 onwards. It feeds into the Looker Studio 2024 ECF Registration Dashboard and informs Mentor metrics. Induction records outside of the registration window are excluded because they don't reflect activity that occurred in registration. There are three marts that inform participants registerred within the annual registration windows: ECT_inductions, Mentor_inductions and Participant_inductions. There are three as they have different fields and cannot be amalgamated into a single mart. They utilise a similar structure logic and are dependent on induction records",
```
Comment (AntPage1) [inline definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_mentor_inductions.sqlx:10]: This table description does not contain the same distinction as the ls_ecf_reg_ect_inductions table description regarding May to October induction dates.

File: definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_mentor_inductions.sqlx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  mentors.mentor_completion_date AS mentor_completion_date,
+  mentors.mentor_completion_reason AS mentor_completion_reason,
+  --  # This translates the null values from the field into falses to permit the use of booleans in dashboards#
+  CASE
+    WHEN cohort_rolled_over_21_24 IS NULL THEN FALSE
+    ELSE cohort_rolled_over_21_24
+END
+  AS mentor_21,
+FROM
+  `dataform.ecf_inductions` mentors
```
Comment (AntPage1) [inline definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_mentor_inductions.sqlx:82]: Should be using the Javascript `${ref()}` statement

File: definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_participant_inductions.sqlx [hunk 4]
```diff
@@ -0,0 +1,114 @@
+config {
+    type: "table",
+    assertions: {
+        uniqueKey: ["induction_record_id"]
+    },
+    bigquery: {
+        partitionBy: "DATE(created_at)",
+        clusterBy: ["cohort"]
+    },
+    description: "This mart pulls together all ECT & Mentor induction records generated within the registration periods from 2022 onwards. It feeds into the Looker Studio 2024 ECF Registration Dashboard and informs combined participant graphs. Induction records outside of the registration window are excluded because they don't reflect activity that occurred in registration. There are three marts that inform participants registerred within the annual registration windows: ECT_inductions, Mentor_inductions and Participant_inductions. There are three as they have different fields and cannot be amalgamated into a single mart. They utilise a similar structure logic and are dependent on induction records",
```
Comment (AntPage1) [inline definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_participant_inductions.sqlx:10]: May to October period is not referenced in this table description but is in ls_ecf_reg_ect_inductions.

File: definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_ect_inductions.sqlx (hunk 1 above)
Comment (consult1815) [reply definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_ect_inductions.sqlx:53]: Sorry about that! Have added the Javascript across all marts.

File: definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_mentor_inductions.sqlx (hunk 2 above)
Comment (consult1815) [reply definitions/marts/looker_studio_marts/ecf/ls_ecf_reg_mentor_inductions.sqlx]: Thanks for flagging! Have added it in to all relevant marts.

**Review by AntPage1 — APPROVED** [review]
All looks good 👍

**Pull request merged**